aten/src/ATen/native/cuda/UpSampleBicubic2d.cu

```diff
@@ -1,25 +1,335 @@
 #include <ATen/ATen.h>
+#include <ATen/AccumulateType.h>
 #include <ATen/NativeFunctions.h>
-#include <ATen/LegacyTHFunctions.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/Utils.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/native/cuda/UpSample.cuh>
 
 namespace at {
 namespace native {
+namespace {
+
+template <typename scalar_t, typename accscalar_t>
+C10_LAUNCH_BOUNDS_1(1024)
```

Comment:
I've approved many of the ROCm PRs that added (possibly unjustified) launch bounds, because in the interest of expediency it's helpful to hardcode assumptions about the AMD hardware in question, but I'm not sure this launch bound is justified on CUDA. If we look at the launch site:

```
  const int64_t max_threads =
      at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;

        upsample_bicubic2d_out_frame<scalar_t, accscalar_t>
            <<<(num_output_elements + max_threads - 1) / max_threads,
               max_threads,
               0,
               stream>>>
```

Although it is difficult to find a device for which `maxThreadsPerBlock > 1024`, to be strictly correct, you should clamp the kernel launch to 1024, if you want to keep these bounds.

My preferred series of resolutions:
1. Check if `const&` on `PackedTensorAccessor` solves the register problem
2. If it doesn't, fix the launch code to clamp to 1024

Reply:
According to @skrah, fixing the PackedTensorAccessor problem solves the register problem. So I would prefer to remove the launch bounds entirely.

Reply:
I haven't looked in detail, but even if it's ok w/o launch bounds on bigger cards, removing launch_bounds is probably going to break jetson (see multpile issues where someone is reporting code breaking on jetson, and the fix is to add launch_bounds), and also future compiler versions can become more register hungry (it happened before).

Reply:
Thank you; @xmnlab, you should listen to Natalia about this. In that case we need to fix the kernel launch.

Reply:
sounds good, thanks @ngimel @ezyang !